plasmapy/physics/transport.py

```diff
@@ -107,98 +168,745 @@ def Coulomb_logarithm(T, n_e, particles, V=None):
+    # deBroglie wavelength
+    lambdaDB = hbar / (2 * reduced_mass * V)
+    # distance of closest approach in 90 degree Coulomb collision
+    bPerp = b_perp(T=T,
+                    particles=particles,
+                    V=V)
+    # obtaining minimum and maximum impact parameters depending on which
+    # method is requested
+    if method == "classical":
+        bmax = lambdaDe
+        # Coulomb-style collisions will not happen for impact parameters
+        # shorter than either of these two impact parameters, so we choose
+        # the larger of these two possibilities. That is, between the
+        # deBroglie wavelength and the distance of closest approach.
+        if bPerp > lambdaDB:
+            bmin = bPerp
+        else:
+            bmin = lambdaDB
+    elif method == "GMS-1":
+        # 1st method listed in Table 1 of reference [1]
+        # This is just another form of the classical Landau-Spitzer
+        # approach, but bmin is interpolated between the deBroglie
+        # wavelength and distance of closest approach.
+        bmax = lambdaDe
+        bmin = (lambdaDB ** 2 + bPerp ** 2) ** (1 / 2)
+    elif method == "GMS-2":
+        # 2nd method listed in Table 1 of reference [1]
+        # Another Landau-Spitzer like approach, but now bmax is also
+        # being interpolated. The interpolation is between the Debye
+        # length and the ion sphere radius, allowing for descriptions
+        # of dilute plasmas.
+        bmax = (lambdaDe ** 2 + ionRadius ** 2) ** (1 / 2)
+        bmin = (lambdaDB ** 2 + bPerp ** 2) ** (1 / 2)
+    elif method == "GMS-3":
+        # 3rd method listed in Table 1 of reference [1]
+        # same as GMS-1, but not Lambda has a clamp at Lambda_min = 2
+        # where Lambda is the argument to the Coulomb logarithm.
+        bmax = lambdaDe
+        bmin = (lambdaDB ** 2 + bPerp ** 2) ** (1 / 2)
+    elif method == "GMS-4":
+        # 4th method listed in Table 1 of reference [1]
+        bmax = lambdaDe
+        bmin = (lambdaDB ** 2 + bPerp ** 2) ** (1 / 2)
+    elif method == "GMS-5":
+        # 5th method listed in Table 1 of reference [1]
+        bmax = (lambdaDe ** 2 + ionRadius ** 2) ** (1 / 2)
+        bmin = bPerp
+    elif method == "GMS-6":
+        # 6th method listed in Table 1 of reference [1]
+        bmax = (lambdaDe ** 2 + ionRadius ** 2) ** (1 / 2)
+        bmin = (lambdaDB ** 2 + bPerp ** 2) ** (1 / 2)
     else:
-        _check_relativistic(V, 'Coulomb_logarithm', betafrac=0.8)
+        raise ValueError(f"Method {method} not found!")
+    return (bmin, bmax)
 
-    # The first possibility is that the inner impact parameter
-    # corresponds to a deflection of 90 degrees, which is valid when
-    # classical effects dominate.
 
-    b_perp = charges[0] * charges[1] / (4 * pi * eps0 * reduced_mass * V**2)
+@check_quantity({"T": {"units": u.K, "can_be_negative": False},
+                 "n_e": {"units": u.m**-3}
+                 })
+def collision_frequency(T,
```

Comment:
This got pretty long and I'd like to go to sleep soonish. I'll get back to this file later!

Reply:
🛌 💤 Sleep is good. I can highly recommend it. Especially as this is only going to get longer. 😀 👿